### ISSUE TITLE
Slowness reported with ejections

### ISSUE THREAD

**Issue by IceRaptor** [BattletechModders/PanicSystem#2]
Xeryx reports that some users are reporting 'frame drops' during the ejection system:

```
Hey Frosty, I am not using logging and experiencing frame drops from the ejections system, I am not sure if there is an issue between CAC, Ejection, or a combination.  There is a slight delay or stutter before the pilot info rolls up in the floatie messages.  I am not using any logging and experiencing this. I can understand when logging it doing it but it souldn't be doing it when logging is set to false
```

Possibly related to deferred harmony patches that occur in the middle of the attack sequences, like at 
https://github.com/BattletechModders/PanicSystem/blob/cbbf91c5053e66acb88018ac084e720bd6731ee1/Patches/AttackStackSequence.cs#L182
?

**Comment by fmnatic** [comment]
Refactored the harmony patches. under test.

**Comment by xeryx01** [comment]
Playing with your release of 4.0 now not DZ's.  I did find that there may be a couple of interrelated issues. 1) CAC language settings need to match user language. 2) Battletech Performance fix is really screwing up battles. Maybe there are some overlapping patches, but there are definitely some compatibility issues with BTPF since it has not been updated. I would recommend doing some testing.  I have also noted dloss of the Floatie Icons on many occasions and performance seems to go down when this happens and maybe related to CAC

**Comment by fmnatic** [comment]
xeryx, from what i've fixed, the PanicSystem was responsible for slowness with vehicle ejections. Are you playing with combat log off ? and when does the slowness happen in game? I've broadly being looking at stalls that pause/stutter the animations in game across mods, so any specific scenarios you can provide, will help.

**Comment by xeryx01** [comment]
@fmnatic Yes, when I test FPS and stuff I test with all logging off. I however do not play with vehicle ejections? I do use CAC in XAI and it computes a lot when weapons are incoming. Since you changed when the patches are applied on 4.1 it does seem much better and reduced frame drops.  The main time that I do notice a slowdown or frame drop is when Ejection actually happens.  Not sure if is due to accessing the pilot picture or phrase box, but I think having an option to turn that off, and just use the floatie, in the mod.json would also help for peeps with slow systems. I have noticed some minor issues with both CAC and the Panic System and it may be a conflict between the two or a common problem, but I have noticed that the floatie Icons disappear quite a bit.  This issue has been really hard to track down because I don't seem to get any real errors about it.  Maybe it is a GUID error but all the "coding" stuff is beyond my skillset.  This is an intermittent problem and doesn't seem to affect performance.

**Comment by fmnatic** [comment]
@xeryx01  My guess is the floatie icons don't disappear. They actually are one over the other i.e. overlap, or just get removed when there is a high rate of floaties. This is based on some other BT UI, whose performance and quirks i was looking into, so its just a guess. 
I'm on Discord as Fmnatic, currently i'm mostly tracking down BT performance issues causing framedrops or breaks in animation.

**Comment by xeryx01** [comment]
@fmnatic add me as a friend ands we cna chat more!  Xeryx#1261

**Comment by fmnatic** [comment]
this was fixed in 4.1.0
